tools/package-client.js

```diff
@@ -390,6 +408,91 @@ exports.handlePackageServerConnectionError = function (error) {
+// descriptions and the Git URL to new values.
+//
+// options:
+// - packageSource: the packageSource for this package.
+// - readmeInfo: null, or an object containing docs information for this package.
+// - connection: the open, logged-in connection over which we should talk to the
+//   package server. DO NOT CLOSE this connection here.
+//
+// Return true on success and an error code otherwise.
+exports.updatePackageMetadata = function (options) {
+  buildmessage.assertInJob();
+
+  var packageSource = options.packageSource;
+  var conn = options.connection;
+  var readmeInfo = options.readmeInfo;
+
+  var name = packageSource.name;
+  var version = packageSource.version;
+
+  if (! version) {
+    buildmessage.error(
+      "Package cannot be updated because it doesn't have a version");
+    return;
+  }
+
+  // For now, documentation is optional on the client, so we have to give people
+  // a way to remove it with 'documentation: null'.
+  if (! readmeInfo) {
+    readmeInfo = generateBlankReadme();
+  }
+
+  // This function is the only way to change README contents after
+  // publication. Since we are automatically migrating a lot of READMEs, we want
+  // to make it easy for people to change them. As such, we are unusually lax
+  // about missing information or making sure that the source matches the
+  // server.
+  // XXX: Tighten these restrictions in later versions.
+  var git = packageSource.metadata.git;
+  var summary = packageSource.metadata.summary;
+  var dataToUpdate = { };
+  if (git !== null) {
+    dataToUpdate["git"] = git;
+  }
+  dataToUpdate["longDescription"] = readmeInfo.excerpt;
+  // Since you can't set your summary to an empty string, we don't need to be as
+  // explicit about the possibility of an empty summary.
+  if (summary) {
+    dataToUpdate["description"] = summary;
+  }
+
+  // Check that the metadata fits under the established limits, and give helpful
+  // feedback.
+  if (dataToUpdate["description"] &&
+      dataToUpdate["description"].length > 100) {
+    buildmessage.error("Summary must be under 100 chars.");
+    return;
+  }
+  if (dataToUpdate["longDescription"].length > 1500) {
+    buildmessage.error(
+      "Documentation snippet is too long. Meteor uses the section of",
+      "the markdown documentation file, between the first and second",
+      "headings, for the long-form package description. That section",
+      "must be less than 1500 characters long.");
+    return;
+  }
+
+  // Update the general metadata.
+  var versionIdentifier = { packageName: name, version: version };
+  buildmessage.enterJob('updating metadata', function () {
+    callPackageServerBM(
+      conn, "changeVersionMetadata", versionIdentifier, dataToUpdate);
+  });
+
+  // Upload the new Readme.
+  buildmessage.enterJob('uploading documentation', function () {
+    var uploadInfo =
+          callPackageServerBM(conn, "createReadme", versionIdentifier);
```

Comment:
I see some other added returns, but not one between callPackageServerBM and uploadFile here.